Build fails on  JDK17

When running `./gradlew check` I get the following failure 

```text
> Task :tribune-core:compileKotlin FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tribune-core:compileKotlin'.
> Inconsistent JVM-target compatibility detected for tasks 'compileJava' (17) and 'compileKotlin' (11).
  
  Consider using JVM Toolchain: https://kotl.in/gradle/jvm/toolchain
  Learn more about JVM-target validation: https://kotl.in/gradle/jvm/target-validation 
```

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -23,6 +23,9 @@ allprojects {
 
    apply(plugin = "org.jetbrains.kotlin.jvm")
 
+   java {
+      sourceCompatibility = JavaVersion.VERSION_11
+   }
    tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
       kotlinOptions {
          jvmTarget = "11"

